IBF-Typhoon-model/lib/pipeline.py

```diff
@@ -0,0 +1,314 @@
+#!/bin/sh
+
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Oct 31 16:01:00 2020
+
+@author: ATeklesadik
+"""
+import sys
+import os
+from datetime import datetime, timedelta
+from sys import platform
+import subprocess
+import logging
+import traceback
+
+import pandas as pd
+from pybufrkit.decoder import Decoder
+import numpy as np
+from geopandas.tools import sjoin
+import geopandas as gpd
+import click
+
+# Set up logger
+logging.root.handlers = []
+logging.basicConfig(format='%(asctime)s : %(levelname)s : %(message)s', level=logging.DEBUG, filename='ex.log')
+# set up logging to console
+console = logging.StreamHandler()
+console.setLevel(logging.INFO)
+# set a format which is simpler for console use
+formatter = logging.Formatter('%(asctime)s : %(levelname)s : %(message)s')
+console.setFormatter(formatter)
+logging.getLogger("").addHandler(console)
+
+decoder = Decoder()
+#path='C:/Users/ATeklesadik/OneDrive - Rode Kruis/Documents/Typhoon-Impact-based-forecasting-model/IBF-Typhoon-model/'
+#path='/home/fbf/'
+path = './'
+#%%
+sys.path.insert(0, path+'lib')
+os.chdir(path)
+from settings import fTP_LOGIN, fTP_PASSWORD, uCL_USERNAME, uCL_PASSWORD
+from climada.hazard import Centroids, TropCyclone,TCTracks
+from climada.hazard.tc_tracks import estimate_roci,estimate_rmw
+from climada.hazard.tc_tracks_forecast import TCForecast
+from utility_fun import track_data_clean,Check_for_active_typhoon,Sendemail,ucl_data
+from utility_fun import Rainfall_data
+
+
+@click.command()
+@click.option('--path', default='./', help='main directory')
+@click.option('--remote_dir_', default='20210421120000', help='remote directory for ECMWF forecast data') 
+@click.option('--active_typhoon', default='SURIGAE',help='name for active typhoon')
```

Comment:
How do you want the `remote_dir` and `active_typhoon` command line options to be used? I think something that would be useful (and save a lot of typing) would be just to have a `--debug` flag option, in which case the pipeline would automatically Surigae. That way, if it runs and there is no active typhoon, it will just stop (I assume this is what you wan to have happen operationally). Then I don't think you would need the `remote_dir` option. 

Also it might make sense to rename this to `typhoon-name`, and perhaps it could be used in the case of  multiple simultaneous typhoons (does that ever happen?) if you only want to run for one

Reply:
i lie the  debug idea but i think the online repo of ecmwf is available only for few months, in the future we have to change typhoon names /directory

Reply:
Do you mean that in the future we will always have to explicitly provide a typhoon name and directory?